File: problems/p1848_minimum_distance_to_the_target_element/solution.py
# 1848. Minimum Distance to the Target Element
# https://leetcode.com/problems/minimum-distance-to-the-target-element/

from typing import List


class Solution:
    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        index = -1;
        for i, v in enumerate(nums):
            if v == target:
                if abs(i-start) < abs(index-start) or index == -1:
                    index = i;
        return abs(index-start);

File: problems/p1848_minimum_distance_to_the_target_element/solution.cpp
// 1848. Minimum Distance to the Target Element
// https://leetcode.com/problems/minimum-distance-to-the-target-element/

#include <algorithm>
#include <numeric>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    int getMinDistance(const vector<int>& nums, int target, int start) {
        int best = nums.size();
        for (int i = 0; i < static_cast<int>(nums.size()); i++) {
            if (nums[i] == target) {
                best = min(best, abs(i - start));
            }
        }
        return best;
    }
};
